tests/prototypes/workflows.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from typing import List, NewType
+
+import scipp as sc
+
+from beamlime.constructors import Factory, ProviderGroup
+
+from .parameters import HistogramBinSize, NumPixels
+
+# Coordinates
+PixelID = NewType("PixelID", sc.Variable)
+EventTimeOffset = NewType("EventTimeOffset", sc.Variable)
+Length = NewType("Length", sc.Variable)
+WaveLength = NewType("WaveLength", sc.Variable)
+
+# Constants
+CoordTransformGraph = NewType("CoordTransformGraph", dict)
+
+# Generated/Calculated
+Events = NewType("Events", List[sc.DataArray])
+MergedData = NewType("MergedData", sc.DataArray)
+PixelIDEdges = NewType("PixelIDEdges", sc.Variable)
+PixelGrouped = NewType("PixelGrouped", sc.DataArray)
+ReducedData = NewType("ReducedData", sc.DataArray)
+Histogrammed = NewType("Histogrammed", sc.DataArray)
+
+
+def provide_coord_transform_graph() -> CoordTransformGraph:
+    c_a = sc.scalar(0.00001, unit='m')
+    c_b = sc.scalar(0.1, unit='m')
+    c_c = sc.scalar(1, unit='1e-3m^2/s')
+    return CoordTransformGraph(
+        {
+            'L1': lambda pixel_id: (pixel_id * c_a) + c_b,
+            'L2': lambda pixel_id: (pixel_id * c_a) + c_b,
+            'L': lambda L1, L2: L1 + L2,
+            'wavelength': lambda event_time_offset, L: (c_c * event_time_offset / L).to(
+                unit='angstrom'
+            ),
+        }
+    )
+
+
+def workflow_script(
+    da_list: Events, num_pixels: NumPixels, histogram_bin_size: HistogramBinSize
+) -> Histogrammed:
+    merged = sc.concat(da_list, dim='event')
+    pixel_ids = sc.arange(dim='pixel_id', start=0, stop=num_pixels)
+    binned = merged.group(pixel_ids)
+
+    graph = provide_coord_transform_graph()
+
+    transformed = binned.transform_coords(['L', 'wavelength'], graph=graph)
+    return transformed.hist(wavelength=histogram_bin_size).sum('L')
```

Comment:
For instrument like DREAM we need thousands of bins. This will run out of memory, instead we will have to use something like

```python
return transformed.bins.concat('L').hist(wavelength=histogram_bin_size)
```